### DISCUSSION TITLE
Examples are not compiling for me...

### DISCUSSION THREAD

**Discussion by jhaimerl1** [lathoub/Arduino-AppleMIDI-Library#150] [Q&A] [answered]
I have an arduino uno rev 3 with an ethernet shield 2.  I've created some nice little programs for my arduino using a 5-pin midi connector, but ultimately I'd like to connect it with an ethernet cable to my switch and let it to join up to my Macbook Pro Apple MIDI network (Network Session 1, port 5004) with some other Apps.  
I thought I'd start with the examples here. Unfortunately I cant get any of the examples to compile. I see that changes were made a couple weeks ago with AM_DBG stuff and that's what seems to be giving me the issue. I tried stepping back from the current version 3.2.0 to 3.1.2, but that didn't help.  I also noticed some notes in github referring to 3.3, but I'm not seeing that anywhere.

Can anyone point me in the right direction to get this up and running?
Thanks!
(I'm very much a newbie on Arduino and Github, so sorry if this seems basic)

#include <Ethernet.h>
#define SerialMon Serial
#define USE_DIRECTORY
#include <AppleMIDI.h>

// Enter a MAC address for your controller below.
// Newer Ethernet shields have a MAC address printed on a sticker on the shield
byte mac[] = { 0xA8, 0x61, 0x0A, 0xAE, 0x96, 0x34 };

unsigned long t1 = millis();
int8_t isConnected = 0;

APPLEMIDI_CREATE_DEFAULTSESSION_INSTANCE();

// -----------------------------------------------------------------------------
//
// -----------------------------------------------------------------------------
void setup()
{
  AM_DBG_SETUP(115200);
  AM_DBG(F("Booting"));

  if (Ethernet.begin(mac) == 0) {
    AM_DBG(F("Failed DHCP, check network cable & reboot"));
    for (;;);
  }

  AppleMIDI.directory.push_back(IPAddress(192, 168, 1, 63));
  AppleMIDI.directory.push_back(IPAddress(192, 168, 1, 66));
  //  AppleMIDI.whoCanConnectToMe = APPLEMIDI_NAMESPACE::None;
  AppleMIDI.whoCanConnectToMe = APPLEMIDI_NAMESPACE::OnlyComputersInMyDirectory;
  //  AppleMIDI.whoCanConnectToMe = APPLEMIDI_NAMESPACE::Anyone;

  AM_DBG(F("OK, now make sure you an rtpMIDI session that is Enabled"));
  AM_DBG(F("Add device named Arduino with Host"), Ethernet.localIP(), "Port", AppleMIDI.getPort(), "(Name", AppleMIDI.getName(), ")");
  AM_DBG(F("Select and then press the Connect button"));
  AM_DBG(F("Then open a MIDI listener and monitor incoming notes"));

  MIDI.begin();

  // Stay informed on connection status
  AppleMIDI.setHandleConnected([](const APPLEMIDI_NAMESPACE::ssrc_t & ssrc, const char* name) {
    isConnected++;
    AM_DBG(F("Connected to session"), ssrc, name);
  });
  AppleMIDI.setHandleDisconnected([](const APPLEMIDI_NAMESPACE::ssrc_t & ssrc) {
    isConnected--;
    AM_DBG(F("Disconnected"), ssrc);
  });

  MIDI.setHandleNoteOn([](byte channel, byte note, byte velocity) {
    AM_DBG(F("NoteOn"), note);
  });
  MIDI.setHandleNoteOff([](byte channel, byte note, byte velocity) {
    AM_DBG(F("NoteOff"), note);
  });

  AM_DBG(F("Sending MIDI messages every second"));
}

// -----------------------------------------------------------------------------
//
// -----------------------------------------------------------------------------
void loop()
{
  // Listen to incoming notes
  MIDI.read();

  // send a note every second
  // (dont cáll delay(1000) as it will stall the pipeline)
  if ((isConnected > 0) && (millis() - t1) > 1000)
  {
    t1 = millis();

    byte note = random(1, 127);
    byte velocity = 55;
    byte channel = 1;

    MIDI.sendNoteOn(note, velocity, channel);
    //    MIDI.sendNoteOff(note, velocity, channel);
  }
}

<img width="936" alt="Screen Shot 2022-09-11 at 3 30 42 PM" src="https://user-images.githubusercontent.com/112823025/189545802-9ccc3ca6-6ce9-419f-afa9-072a514c7339.png">

/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino: In function 'void setup()':
AVR_Directory.ino:21:3: error: 'AM_DBG_SETUP' was not declared in this scope
   AM_DBG_SETUP(115200);
   ^~~~~~~~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:21:3: note: suggested alternative: 'DBG_SETUP'
   AM_DBG_SETUP(115200);
   ^~~~~~~~~~~~
   DBG_SETUP
AVR_Directory.ino:22:3: error: 'AM_DBG' was not declared in this scope
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:22:3: note: suggested alternative: 'ACBG'
   AM_DBG(F("Booting"));
   ^~~~~~
   ACBG
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino: In lambda function:
AVR_Directory.ino:45:5: error: 'AM_DBG' is not captured
     AM_DBG(F("Connected to session"), ssrc, name);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:43:33: note: the lambda has no capture-default
   AppleMIDI.setHandleConnected([](const APPLEMIDI_NAMESPACE::ssrc_t & ssrc, const char* name) {
                                 ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:22:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino: In lambda function:
AVR_Directory.ino:49:5: error: 'AM_DBG' is not captured
     AM_DBG(F("Disconnected"), ssrc);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:47:36: note: the lambda has no capture-default
   AppleMIDI.setHandleDisconnected([](const APPLEMIDI_NAMESPACE::ssrc_t & ssrc) {
                                    ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:22:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino: In lambda function:
AVR_Directory.ino:53:5: error: 'AM_DBG' is not captured
     AM_DBG(F("NoteOn"), note);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:52:25: note: the lambda has no capture-default
   MIDI.setHandleNoteOn([](byte channel, byte note, byte velocity) {
                         ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:22:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino: In lambda function:
AVR_Directory.ino:56:5: error: 'AM_DBG' is not captured
     AM_DBG(F("NoteOff"), note);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:55:26: note: the lambda has no capture-default
   MIDI.setHandleNoteOff([](byte channel, byte note, byte velocity) {
                          ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:22:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
Multiple libraries were found for "Ethernet.h"
 Used: /Users/josephhaimerl/Documents/Arduino/libraries/Ethernet
 Not used: /Applications/Arduino.app/Contents/Java/libraries/Ethernet
exit status 1
'AM_DBG_SETUP' was not declared in this scope

**Comment by lathoub** [comment]
Try to download version 3.3 from GitHub directly (**not** using the Arduino library manager)

**Reply by jhaimerl1** [reply]
Sorry for being a newb.  I don't see where to download it.  Only see something on right side that says Latest Release 3.2.0

<img width="1202" alt="Screen Shot 2022-09-11 at 3 56 52 PM" src="https://user-images.githubusercontent.com/112823025/189546630-cfe57302-17df-447a-9dae-26bb6a903457.png">

**Comment by lathoub** [comment]
Under the ‘code’ dropdown

(No worries, all of us were newbs at some stage - also pick on markdown, how to annotate a git posting) - good luck

**Reply by jhaimerl1** [reply]
Yes. I just found that and downloaded the zip... from there I selected  Sketch -> Include Library -> Add .ZIP Library and used the ZIP file in my download... I tried compiling and get same errors.

**Reply by jhaimerl1** [reply]
Arduino: 1.8.19 (Mac OS X), Board: "Arduino Uno"











/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino: In function 'void setup()':
AVR_Directory.ino:20:3: error: 'AM_DBG_SETUP' was not declared in this scope
   AM_DBG_SETUP(115200);
   ^~~~~~~~~~~~
AVR_Directory.ino:21:3: error: 'AM_DBG' was not declared in this scope
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:21:3: note: suggested alternative: 'ACBG'
   AM_DBG(F("Booting"));
   ^~~~~~
   ACBG
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino: In lambda function:
AVR_Directory.ino:44:5: error: 'AM_DBG' is not captured
     AM_DBG(F("Connected to session"), ssrc, name);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:42:33: note: the lambda has no capture-default
   AppleMIDI.setHandleConnected([](const APPLEMIDI_NAMESPACE::ssrc_t & ssrc, const char* name) {
                                 ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:21:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino: In lambda function:
AVR_Directory.ino:48:5: error: 'AM_DBG' is not captured
     AM_DBG(F("Disconnected"), ssrc);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:46:36: note: the lambda has no capture-default
   AppleMIDI.setHandleDisconnected([](const APPLEMIDI_NAMESPACE::ssrc_t & ssrc) {
                                    ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:21:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino: In lambda function:
AVR_Directory.ino:52:5: error: 'AM_DBG' is not captured
     AM_DBG(F("NoteOn"), note);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:51:25: note: the lambda has no capture-default
   MIDI.setHandleNoteOn([](byte channel, byte note, byte velocity) {
                         ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:21:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino: In lambda function:
AVR_Directory.ino:55:5: error: 'AM_DBG' is not captured
     AM_DBG(F("NoteOff"), note);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:54:26: note: the lambda has no capture-default
   MIDI.setHandleNoteOff([](byte channel, byte note, byte velocity) {
                          ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory.ino/AVR_Directory.ino.ino:21:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
Multiple libraries were found for "Ethernet.h"
 Used: /Users/josephhaimerl/Documents/Arduino/libraries/Ethernet
 Not used: /Applications/Arduino.app/Contents/Java/libraries/Ethernet
Multiple libraries were found for "AppleMIDI.h"
 Used: /Users/josephhaimerl/Documents/Arduino/libraries/AppleMIDI
 Not used: /Users/josephhaimerl/Documents/Arduino/libraries/Arduino-AppleMIDI-Library-master
exit status 1
'AM_DBG_SETUP' was not declared in this scope


This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Reply by jhaimerl1** [reply]
I started looking at strange directory name in error message and moved files to not have a .ino.ino. Result is similar...



Arduino: 1.8.19 (Mac OS X), Board: "Arduino Uno"




/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino: In function 'void setup()':
AVR_Directory:20:3: error: 'AM_DBG_SETUP' was not declared in this scope
   AM_DBG_SETUP(115200);
   ^~~~~~~~~~~~
AVR_Directory:21:3: error: 'AM_DBG' was not declared in this scope
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino:21:3: note: suggested alternative: 'ACBG'
   AM_DBG(F("Booting"));
   ^~~~~~
   ACBG
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino: In lambda function:
AVR_Directory:44:5: error: 'AM_DBG' is not captured
     AM_DBG(F("Connected to session"), ssrc, name);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino:42:33: note: the lambda has no capture-default
   AppleMIDI.setHandleConnected([](const APPLEMIDI_NAMESPACE::ssrc_t & ssrc, const char* name) {
                                 ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino:21:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino: In lambda function:
AVR_Directory:48:5: error: 'AM_DBG' is not captured
     AM_DBG(F("Disconnected"), ssrc);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino:46:36: note: the lambda has no capture-default
   AppleMIDI.setHandleDisconnected([](const APPLEMIDI_NAMESPACE::ssrc_t & ssrc) {
                                    ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino:21:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino: In lambda function:
AVR_Directory:52:5: error: 'AM_DBG' is not captured
     AM_DBG(F("NoteOn"), note);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino:51:25: note: the lambda has no capture-default
   MIDI.setHandleNoteOn([](byte channel, byte note, byte velocity) {
                         ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino:21:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino: In lambda function:
AVR_Directory:55:5: error: 'AM_DBG' is not captured
     AM_DBG(F("NoteOff"), note);
     ^~~~~~
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino:54:26: note: the lambda has no capture-default
   MIDI.setHandleNoteOff([](byte channel, byte note, byte velocity) {
                          ^
/Users/josephhaimerl/Desktop/Aurduino Stuff/Ethernet MIDI/AVR_Directory/AVR_Directory.ino:21:3: note: '<typeprefixerror>AM_DBG' declared here
   AM_DBG(F("Booting"));
   ^~~~~~
Multiple libraries were found for "Ethernet.h"
 Used: /Users/josephhaimerl/Documents/Arduino/libraries/Ethernet
 Not used: /Applications/Arduino.app/Contents/Java/libraries/Ethernet
Multiple libraries were found for "AppleMIDI.h"
 Used: /Users/josephhaimerl/Documents/Arduino/libraries/AppleMIDI
 Not used: /Users/josephhaimerl/Documents/Arduino/libraries/Arduino-AppleMIDI-Library-master
exit status 1
'AM_DBG_SETUP' was not declared in this scope


This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Comment by lathoub** [comment, marked as answer]
(No need to include the entire error stack.)
Did you remove the old 3.2 version from the library folder?

**Reply by jhaimerl1** [reply]
I found the library folder and deleted everything AppleMIDI... restarted Arduino program again... downloaded zip file again... added library from zip file... now it compiles!  Now on to see if I can do anything with it!  Thanks!!!

**Reply by jhaimerl1** [reply]
OK I have Note On & Off working!

**Reply by lathoub** [reply]
> OK I have Note On & Off working!

good stuff :-) not a newb anymore then - hihi
